pkg/controllers/pod.go

```diff
@@ -127,34 +127,78 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.R
 	}
 
 	if apierrors.IsNotFound(err) {
-		images, ok := r.podImages[req.NamespacedName.String()]
-		if ok {
-			for _, image := range images {
-				if usage, ok := r.imagePlatforms[image]; ok {
-					usage.refcount--
-					for _, platform := range usage.platforms {
-						r.metrics.ImageCount.WithLabelValues(image, platform.OS, platform.Architecture, platform.Variant).Dec()
-						if usage.refcount == 0 {
-							r.metrics.ImageCount.DeleteLabelValues(image, platform.OS, platform.Architecture, platform.Variant)
-						}
-					}
-					if usage.refcount == 0 {
-						delete(r.imagePlatforms, image)
-					}
-				}
-			}
-			delete(r.podImages, req.NamespacedName.String())
-		}
+		r.deleteFromCaches(req.NamespacedName.String())
 		return ctrl.Result{}, nil
 	}
-	if _, ok := r.podImages[req.NamespacedName.String()]; ok || podIsReady(ctx, pod) {
+
+	if !skipCache(pod) && (r.checkImageCached(req.NamespacedName.String()) || podIsReady(ctx, pod)) {
```

Comment:
I'd recommend making this condition explicit. Otherwise, one must dig into the `skipCache(*v1.Pod)` to understand what it d does:

```suggestion
	if !isStatefulSet(pod) && (r.checkImageCached(req.NamespacedName.String()) || podIsReady(ctx, pod)) {
```

Reply:
I assumed there are other reasons to skip the cache, this being just one. But happy to go with a shorter term approach for the time being.

Reply:
https://martinfowler.com/bliki/Yagni.html